Add configure argument to make the libxml2 dependency explicit and
determinisitic.

Upstream-Status: Pending

Signed-off-by: Christopher Larson <chris_larson@mentor.com>

Rebase to 4.3.4
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 configure.ac | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 726c88e..1684df1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -718,7 +718,16 @@ AC_SUBST(BINDSRCDIR)
 
 # We need to find libxml2 if bind was built with support enabled
 # otherwise we'll fail to build omapip/test.c
-AC_SEARCH_LIBS(xmlTextWriterStartElement, [xml2],)
+AC_ARG_WITH(libxml2,
+	AS_HELP_STRING([--with-libxml2], [link against libxml2. this is needed if bind was built with xml2 support enabled]),
+	with_libxml2="$withval", with_libxml2="no")
+
+if test x$with_libxml2 != xno; then
+    AC_SEARCH_LIBS(xmlTextWriterStartElement, [xml2],
+                   [if test x$with_libxml2 != xauto; then
+                        AC_MSG_FAILURE([*** Cannot find xmlTextWriterStartElement with -lxml2 and libxml2 was requested])
+                    fi])
+fi
 
 # OpenLDAP support.
 AC_ARG_WITH(ldap,
-- 
2.8.1

